src/spot_skills/src/spot_skills/scene_handler.py

```diff
@@ -0,0 +1,205 @@
+        while not rospy.is_shutdown():
+            self.broadcast_frames()
+            self.rate_hz.sleep()
+
+    def pose_to_transform(self, pose: Pose) -> Transform:
+        """Convert a geometry_msgs/Pose message to a geometry_msgs/Transform."""
+        translation = Vector3(pose.position.x, pose.position.y, pose.position.z)
+        return Transform(translation, pose.orientation)
+
+    def broadcast_frames(self) -> None:
+        """Broadcast the frames and subframes for all objects in the planning scene."""
+        scene_objects_dict = self._scene.get_objects()
+
+        for obj_id, obj_msg in scene_objects_dict.items():
+            rospy.loginfo(f"Scene contains object with ID '{obj_id}'")
+
+            obj_tf = TransformStamped()
+            obj_tf.header.stamp = rospy.Time.now()
+            obj_tf.header.frame_id = obj_msg.header.frame_id
+
+            obj_tf.child_frame_id = obj_id
+            obj_tf.transform = self.pose_to_transform(obj_msg.pose)
+            self._broadcaster.sendTransform(obj_tf)
+
+            assert len(obj_msg.subframe_names) == len(obj_msg.subframe_poses)
+
+            for subframe_name, subframe_pose in zip(
+                obj_msg.subframe_names,
+                obj_msg.subframe_poses,
+            ):
+                rospy.loginfo(f"Object '{obj_id}' has the subframe '{subframe_name}'")
+                subframe_tf = TransformStamped()
+                subframe_tf.header.stamp = rospy.Time.now()
+                subframe_tf.header.frame_id = obj_id  # Subframe is w.r.t. object
+
+                subframe_tf.child_frame_id = f"{obj_id}/{subframe_name}"
+                subframe_tf.transform = self.pose_to_transform(subframe_pose)
+                self._broadcaster.sendTransform(subframe_tf)
+
+    def create_collision_object_msg(
+        self,
+        object_id: str,
+        object_type: str,
+        mesh: Trimesh,
+        relative_frame: str,
+        object_pose: Pose,
+    ) -> CollisionObject:
+        """Create and return a moveit_msgs/CollisionObject message.
+
+        Note: This method sets the CollisionObject's `operation` field to ADD.
+
+        :param object_id: Unique identifier for the object
+        :param object_type: Type of the object
+        :param mesh: Mesh defining the object's collision geometry
+        :param relative_frame: Frame w.r.t. which the object's pose is defined
+        :param object_pose: Pose for the object in the global frame
+
+        :returns: moveit_msgs/CollisionObject based on the given inputs.
+        """
+        collision_object = CollisionObject()
+        collision_object.header.frame_id = relative_frame
+        collision_object.pose = object_pose  # Subframes are defined relative to this
+
+        collision_object.id = object_id
+
+        # TODO: How do object_recognition_msgs/ObjectType messages work?
+        collision_object.type.key = object_type
```

Comment:
Answer

Reply:
They have a field to specify the database (usage is up to user; I just won't use it)